packages/arb-bridge-peripherals/contracts/tokenbridge/ethereum/EthERC20Bridge.sol

```diff
@@ -34,18 +35,32 @@ contract EthERC20Bridge is L1Buddy {
 
     mapping(address => address) customL2Tokens;
 
-    constructor(IInbox _inbox) public L1Buddy(_inbox) {}
+    address private immutable l2TemplateERC777;
+    address private immutable l2TemplateERC20;
+
+    constructor(
+        IInbox _inbox,
+        address _l2TemplateERC777,
+        address _l2TemplateERC20
+    ) public L1Buddy(_inbox) {
+        l2TemplateERC777 = _l2TemplateERC777;
+        l2TemplateERC20 = _l2TemplateERC20;
+    }
 
     function connectToChain(uint256 maxGas, uint256 gasPriceBid) external payable {
         // Pay for gas
         if (msg.value > 0) {
             inbox.depositEth{ value: msg.value }(address(this));
         }
+        bytes memory deployCode = abi.encodePacked(
+            type(ArbTokenBridge).creationCode,
+            abi.encode(l2TemplateERC20, l2TemplateERC777)
```

Comment:
Should this maybe be encodePacked?

Reply:
I believe this one should be abi.encode, since these are the constructor parameters